lib/galaxy/datatypes/binary.py

```diff
@@ -462,6 +468,52 @@ def to_archive(self, trans, dataset, name=""):
         file_paths.append(dataset.metadata.bam_index.file_name)
         return zip(file_paths, rel_paths)
 
+    def get_chunk(self, trans, dataset, offset=0, ck_size=None):
+        index_file = dataset.metadata.bam_index
+        with pysam.AlignmentFile(dataset.file_name, "rb", index_filename=index_file.file_name) as bamfile:
+            ck_size = 1000  # 1000 lines
+            ck_data = ""
+            line_number = 0
+            if offset == 0:
+                ck_data = bamfile.text
+            for line_number, alignment,  in enumerate(bamfile):
+                if line_number > offset and line_number <= (offset + ck_size):
+                    bamline = alignment.tostring(bamfile)
+                    # Galaxy display each tag as separate column because 'tostring()' funcition put spaces in between each tag of tags column. 
+                    # Below code will remove spaces between each tag. 
+                    bamline_modified = ('\t').join(bamline.split()[:11] + [('').join(bamline.split()[11:])])
+                    ck_data = ck_data +"\n" + bamline_modified
+                elif line_number > (offset + ck_size):
```

Comment:
you can drop this if you implement https://github.com/galaxyproject/galaxy/pull/4279/files#r125709901